projects/schematics/src/shared/utils/file-utils_spec.ts

```diff
@@ -176,20 +203,313 @@ describe('File utils', () => {
     });
   });
 
+  describe('isCandidateForConstructorDeprecation', async () => {
+    it('should return false if the inheritance condition is not satisfied', () => {
+      const content = `
+      import { Store } from '@ngrx/store';
+      import { StateWithProcess, StateWithUser } from '@spartacus/core';
+      export class InheritingService {
+        constructor(_store: Store<StateWithUser | StateWithProcess<void>>) {}
+      }
+      `;
+      const source = ts.createSourceFile(
+        'xxx.ts',
+        content,
+        ts.ScriptTarget.Latest,
+        true
+      );
+
+      expect(
+        isCandidateForConstructorDeprecation(source, USER_ADDRESS_SERVICE, [])
+      ).toEqual(false);
+    });
+    it('should return false if the imports condition not satisfied', () => {
+      const content = `
+      import { StateWithProcess, StateWithUser, UserAddressService } from '@spartacus/core';
+      export class InheritingService extends UserAddressService {
+        constructor(_store: Store<StateWithUser | StateWithProcess<void>>) {}
+      }
+      `;
+      const source = ts.createSourceFile(
+        'xxx.ts',
+        content,
+        ts.ScriptTarget.Latest,
+        true
+      );
+      const parameterClassTypes: ClassType[] = [
+        { className: STORE, importPath: NGRX_STORE },
+      ];
+
+      expect(
+        isCandidateForConstructorDeprecation(
+          source,
+          USER_ADDRESS_SERVICE,
+          parameterClassTypes
+        )
+      ).toEqual(false);
+    });
+    it('should return false if the constructor condition is not satisfied', () => {
+      const content = `
```

Comment:
Can we go with consts before tests (as in `update-cms-component-state_spec.ts`)? Just for more consistency